src/TodosContext.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { useMemo, useState } from 'react';
+import { Status } from './types/Status';
+import { TodosContextType } from './types/TodosContext';
+import { ErrorMessage } from './types/ErrorMessage';
+import { Todo } from './types/Todo';
+
+export const TodosContext = React.createContext<TodosContextType>({
+  todos: [],
+  setTodos: () => {},
+  visibleTodos: [],
+  filter: Status.ALL,
+  setFilter: () => { },
+  isChecked: false,
+  setIsChecked: () => {},
+  errorMessage: ErrorMessage.DEFAULT,
+  setErrorMessage: () => {},
+  isProcessing: [],
+  setIsProcessing: () => {},
+});
+
+type Props = {
+  children: React.ReactNode;
+};
+
+export const TodoProvider: React.FC<Props> = ({ children }) => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isChecked, setIsChecked] = useState<boolean>(false);
+  const [filter, setFilter] = useState(Status.ALL);
+  const [errorMessage, setErrorMessage] = useState(ErrorMessage.DEFAULT);
+  const [isProcessing, setIsProcessing] = useState<number[]>([]);
+
+  const visibleTodos = useMemo(() => {
+    if (filter === Status.ACTIVE) {
+      return todos.filter(todo => !todo.completed);
+    }
+
+    if (filter === Status.COMPLETED) {
```

Comment:
Use [`switch case`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch) in cases where you constantly compare one value with others